docs/python-examples/fixtures/conftest.py

```diff
@@ -0,0 +1,102 @@
+import logging
+
+import pytest
+
+import ocs
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+@pytest.fixture(scope='class')
+def storage_class(request):
+    """
+    Storage class fixture
+
+    Returns:
+        StorageClass: object of storage class
+    """
+    def fin():
+        if not sc.is_deleted:
+            sc.delete()
+
+    request.addfinalizer(fin)
+
+    sc_name = ocs.get_random_name('storage_class')
+    logger.info(f"Creating storage class: {sc_name}")
+    sc = ocs.StorageClass(sc_name)
+    return sc
+
+
+@pytest.fixture(scope='class')
+def cls_pvc(request, storage_class):
+    """
+    PVC fixture
+
+    Returns:
+        PVC: object of PVC class
+    """
+    def fin():
+        if not pvc.is_deleted:
+            pvc.delete()
+
+    request.addfinalizer(fin)
+    pvc_name = ocs.get_random_name('pvc')
+    logger.info(f"Creating pvc: {pvc_name}")
+    pvc = ocs.PVC(pvc_name, storage_class, some_parameter="Default value")
+    return pvc
+
+
+@pytest.fixture(scope='class')
+def pvc_factory(request, storage_class):
+    """
+    Fixture factory for creating pvcs.
```

Comment:
Not a big fan of this due to (1), It adds complexity for many qe folks on how to use it 
(2) can be easily handled at test level without need for this.

Reply:
It _can_ be handled at the test level but the point is that by utilizing these fixtures the tests no longer have to care about cleaning up resources OR creating that resources dependencies as well, and we don't need to ensure that tests are doing so properly. By using the factory fixture they can dictate when they create resources and let the finalizer clean up correctly based on scope.

I'm not seeing the added complexity, usage is actually quite simple:

```
def my_test(pvc_factory):
    pvc = pvc_factory()
    # now the test can do whatever it wants with this pvc object and not care about cleaning up
```

Reply:
Works for me :)